vendor/zkllvm-metacraft-circuits/src/circuit_utils/circuit_byte_utils.h

```diff
@@ -81,22 +101,29 @@ namespace circuit_byte_utils {
         return bytes;
     }
 
-    template<typename T, bool LittleEndian = true>
-    T bytes_to_int(const std::array<Byte, sizeof(T)>& bytes) {
+    template<typename T, typename iterator_type, size_t SIZE, bool LittleEndian>
+    T bytes_to_int(iterator_type first_element, iterator_type last_element) {
         static_assert(std::is_integral_v<typename std::remove_reference_t<T>>, "T must be integral");
+        assert_in_executable(first_element + sizeof(T) <= last_element);
+
         T result = 0;
         if constexpr (LittleEndian) {
-            for (int i = sizeof(T) - 1; i >= 0; i--) {
-                result = (result << 8) + bytes[i];
+            for (unsigned int i = sizeof(T); i > 0; i--) {
+                result = (result << 8) + *(first_element + i - 1);
             }
         } else {
             for (unsigned i = 0; i < sizeof(T); ++i) {
-                result = (result << 8) + bytes[i];
+                result = (result << 8) + *(first_element + i);
             }
         }
         return result;
     }
 
+    template<typename T, size_t SIZE = sizeof(T), bool LittleEndian = true>
```

Comment:
Swap SIZE and LittleEndian so that you can change the endianness without passing sizeof(T)

Reply:
Or remove the SIZE argument completely and use sizeof(T) instead

Reply:
Actually it is more convenient to have the size first, because there are use cases where we want to put uint64_t in a 32 bytes buffer and take a uint64_t from a 32 byte buffer.